### ISSUE TITLE
[Bug]: `misc:` not honored when inside a `- file:` 

### ISSUE THREAD

**Issue by mcesar-rlacruz** [Open-CMSIS-Pack/cmsis-toolbox#119]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

In file `*.cproject.yml`, when specifying compile options for a given file with, for example:
```groups:
  - group: App
    files:
      - file: main.cpp
        misc:
          - C-CPP:
              - -flto
```
such options are not honored. Curiously, file `CMakeLists.txt` does contain the flag (`set(CXX_FLAGS_<path_to_main.cpp> "[…] -flto")`), but file `compile_commands.json` does not.

### Expected Behavior

Both files `CMakeLists.txt` AND `compile_command.json` containing the specified literal tool-specific options specified in the `misc:` field of `file:`, thus compiling source file with such options.

### Affected tools

- [ ] cbuild
- [ ] cbuild-gen
- [ ] cpackget
- [ ] csolution
- [ ] packchk
- [ ] svdconv

### Affected operating systems

- [X] Windows amd64
- [ ] Windows arm64
- [ ] Linux amd64
- [ ] Linux arm64
- [ ] MacOS amd64
- [ ] MacOS arm64

### Environment

- Windows 11 Enterprise 22H2
- Visual Studio Code 1.88.1
- Arm Keil Studio Pack 1.18.2

**Comment by jkrech** [comment]
@brondani it is worth checking that this works using the cbuild2cmake generator and lower priority for fixing it for cbuildgen generated CMakeLists.txt.

**Comment by brondani** [comment]
@mcesar-rlacruz Thanks for reporting this.
A possible workaround is to use `cbuild2cmake` as the build manager backend by calling `cbuild` with the `--cbuild2cmake` option.
In such case you should see the specified file option in the generated groups.cmake like this:
```
set_source_files_properties("${SOLUTION_ROOT}/App/main.cpp" PROPERTIES
  COMPILE_OPTIONS -flto
)
```
And ultimately it should appear in the command line as well as in compile_commands.json. Please note `cbuild2cmake` is in alpha quality.

Anyway The fix for cbuildgen is in progress, it will be part of the next release.

**Comment by jkrech** [comment]
Fix will be in CMSIS-Toolbox 2.4.0
